camelot/core.py

```diff
@@ -597,7 +599,7 @@ def to_json(self, path, **kwargs):
         kw = {"orient": "records"}
         kw.update(kwargs)
         json_string = self.df.to_json(**kw)
-        with open(path, "w") as f:
+        with open(path, "w", encoding="utf-8") as f:
```

Comment:
Here, and below I'd use `with open(path, "wb") as f:` and then `f.write(json_string.encode("utf-8"))`. I tested this on your branch locally under both Python 3.7 and 2.7, and seems to work nicely.

Reply:
I have considered that, it should help, but it is ugly and dirty python 2 style legacy. Upgrading to python 3.4 doesn't require any hardware or OS upgrades for Win XP users (an if python was built with MinGW, it would have been possible even for them to have the latest python), *nix OSes users can upgrade even to the latest version. There are no problems with migration of python code, only some native extensions are a problem, but for them to be a problem it is needed that noone have touched them for decades.

IMHO: python 2 should be just dropped. Less packages support it - more users have to upgrade - less sense to support it for the rest of packages. If we wanna get rid of need to support python 2 (not only in this lib, but in the ecosystem, I dislike adapting my PRs for the projects still supporting python 2), we should just drop it.

Reply:
I also think that we should drop Python 3 support, we can set a date on https://github.com/camelot-dev/camelot/issues/39. Till then, we should merge this with the changes @dimitern suggested.

Reply:
I personally don't use python 2. IDK if there is a better solution than the proposed by @dimitern, but I feel like that: I don't remember any projects compatible with both versions having so dirty solution. So I propose the following:

1. declare dropping python 2 immediately, but also establish a deadline after which support of python 2 is dropped unconditionally
2. drop it immediately
3. if someone needs python 2, he fixes the code added with this PR first himself, then we declare return of python 2 support untill the deadline. If any PRs breaks python 2 support, we break it again and ones needing python 2 have to fix them again.